docs/tutorials/custom-id-generator.rst

```diff
@@ -0,0 +1,82 @@
+.. _tutorial-id-generator:
+
+How to define and use a custom ID generator?
+============================================
+
+By default, Kinto uses UUID4 as ``id`` for its records. In some cases, it might
+not be what you want. But you're not out of luck, there is a way to configure
+Kinto to use a different ``id_generator`` than the default one.
+
+Let's define a "name generator", which will output 8 random characters
+(uppercase or lowercase).
+
+.. code-block:: python
+
+    import random
+    import string
+
+    from cliquet.storage import generators
+
+
+    class NameGenerator(generators.Generator):
+
+        regexp = r'^[a-zA-Z0-9-_]{8}$'
```

Comment:
``os.urandom`` is probably not a good idea because it's suppose to be random. From the python docs:

>Return a string of n random bytes suitable for cryptographic use. This function returns random bytes from an OS-specific randomness source. The returned data should be unpredictable enough for cryptographic applications.